app/src/main/java/org/fossasia/openevent/activities/MainActivity.java

```diff
@@ -86,6 +99,23 @@ protected void onCreate(Bundle savedInstanceState) {
 
     }
 
+    public void WiFiDialog(final Activity activity){
+        downloadProgress.setVisibility(View.GONE);
+        AlertDialog.Builder builder = new AlertDialog.Builder(activity);
+        builder.setTitle(getString(R.string.net_unavailable))
+                .setMessage(getString(R.string.turn_on))
+                .setPositiveButton(R.string.action_settings, new DialogInterface.OnClickListener() {
+                    @Override
+                    public void onClick(DialogInterface dialog, int which) {
+                        Intent setNetworkIntent = new Intent(Settings.ACTION_SETTINGS);
+                        startActivity(setNetworkIntent);
+                        activity.finish();
```

Comment:
I tried keeping my network on and then running the app and switching it off while the download is in progress. It gives a Network Error (Socket time-out exception) everytime.

Instead of adding a BroadcastReceiver, why not just add this case ^ to my existent method?